### PR TITLE
Add whitelisting errors functionality to the breaker.

### PR DESCRIPTION
**Pull request by phrmsilva**
Add a functionality for the circuit breaker to ignore and to not open on errors specified by the user.

### CODE REVIEW

**Comment by phrmsilva** [comment]
@morganjbruce @SofianO

**Comment by phrmsilva** [comment]
@ranierorusso

File: lib/cb2/breaker.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class CB2::Breaker
   def initialize(options)
     @service  = options[:service] || "default"
     @strategy = initialize_strategy(options)
+    @whitelist = if options[:whitelist] then options[:whitelist] else [] end
```
Comment (ranierorusso) [inline lib/cb2/breaker.rb]: can also be written as `options[:whitelist] || []`

Comment (ranierorusso) [inline lib/cb2/breaker.rb]: i would try to use a more specific variable name. @unprocessable_exceptions or @exceptions_blacklist or something of the sort.

File: lib/cb2/breaker.rb [hunk 2]
```diff
@@ -16,7 +17,7 @@ def run
       ret = yield
       process_success
     rescue => e
-      process_error
+      if (!(@whitelist.any? { |klass| e.is_a?(klass) })) then process_error end
```
Comment (ranierorusso) [inline lib/cb2/breaker.rb]: move to separate method called, e.g. `error_can_be_processed?(error)`, in order to document what is happening a little better.

Comment (ranierorusso) [inline lib/cb2/breaker.rb]: opt for ruby's `action if clause` syntax rather than `if clause then action end`.

File: lib/cb2/breaker.rb (hunk 1 above)
Comment (ranierorusso) [inline lib/cb2/breaker.rb]: do you think it's worth considering adding validation code to make sure the values passed in to @whitelist are of type Class and inherit from StandardError?

File: spec/breaker_spec.rb [hunk 3]
```diff
@@ -7,6 +7,14 @@
       allow:    false)
   end
 
+  before do
+    # allow(Redis).to receive(:new).and_return(MockRedis.new)
+  end
+
+  after do
+    # allow(Redis).to receive(:new).and_call_original
+  end
```
Comment (ranierorusso) [inline spec/breaker_spec.rb]: if commented out because no longer needed, then please remove these lines.

File: spec/breaker_spec.rb [hunk 4]
```diff
@@ -18,6 +26,40 @@
...
+        6.times {
+          begin
+            breaker.run { raise 'sample-runtime-error' }
+          rescue
+          end
+        }
+        assert_equal 42, breaker.run { 42 }
+      end
+
+      it "does not ignore errors not whitelisted" do
```
Comment (ranierorusso) [inline spec/breaker_spec.rb]: double negative. maybe "processes errors that are not whitelisted." ?

**Comment by phrmsilva** [comment]
@ranierorusso Branch updated to reflect your comments.

**Pull request merged**